smartsim/_core/control/previewrenderer.py

```diff
@@ -0,0 +1,187 @@
+def get_ifname(_eval_ctx: u.F, value: t.List[str]) -> str:
+    if value:
+        for val in value:
+            if "ifname=" in val:
+                output = val.split("=")[-1]
+                return output
+    return ""
+
+
+@pass_eval_context
+def get_dbtype(_eval_ctx: u.F, value: str) -> str:
+    if value:
+        if "-cli" in value:
+            db_type, _ = value.split("/")[-1].split("-", 1)
+            return db_type
+    return ""
+
+
+@pass_eval_context
+def is_list(_eval_ctx: u.F, value: str) -> bool:
+    return isinstance(value, list)
+
+
+def render(
+    exp: "Experiment",
+    manifest: t.Optional[Manifest] = None,
+    verbosity_level: Verbosity = Verbosity.INFO,
+    output_format: Format = Format.PLAINTEXT,
+    active_dbjobs: t.Optional[t.Dict[str, Job]] = None,
+) -> str:
+    """
+    Render the template from the supplied entities.
+    :param experiment: the experiment to be previewed.
+    :type experiment: Experiment
+    :param manifest: the manifest to be previewed.
+    :type manifest: Manifest
+    :param verbosity_level: the verbosity level
+    :type verbosity_level: Verbosity
+    :param output_format: the output format.
+    :type output_format: _OutputFormatString
+    """
+
+    verbosity_level = _check_verbosity_level(Verbosity(verbosity_level))
+
+    loader = jinja2.PackageLoader("templates")
+    env = jinja2.Environment(loader=loader, autoescape=True)
+
+    env.filters["as_toggle"] = as_toggle
+    env.filters["get_ifname"] = get_ifname
+    env.filters["get_dbtype"] = get_dbtype
+    env.filters["is_list"] = is_list
+    env.globals["Verbosity"] = Verbosity
+
+    version = f"_{output_format}"
+    tpl_path = f"preview/base{version}.template"
+
+    _check_output_format(output_format)
+
+    tpl = env.get_template(tpl_path)
+
+    if verbosity_level == Verbosity.INFO:
+        logger.warning(
+            "Only showing user set parameters. Some entity fields are "
+            "truncated. To view truncated fields: use verbosity_level "
+            "'developer' or 'debug.'"
+        )
+
+    rendered_preview = tpl.render(
+        exp_entity=exp,
+        active_dbjobs=active_dbjobs,
+        manifest=manifest,
+        config=CONFIG,
+        verbosity_level=verbosity_level,
+    )
+    return rendered_preview
+
+
+def preview_to_file(content: str, filename: str) -> None:
+    """
+    Output preview to a file if an output filename
+    is specified.
+    """
+    filename = find_available_filename(filename)
+
+    with open(filename, "w", encoding="utf-8") as prev_file:
+        prev_file.write(content)
+
+
+def find_available_filename(filename: str) -> str:
+    """Iterate through potentially unique names until one is found that does
+    not already exist. Return an unused name variation"""
+
+    path = pathlib.Path(filename)
+    candidate_path = pathlib.Path(filename)
+    index = 1
+
+    while candidate_path.exists():
+        candidate_path = path.with_name(f"{path.stem}_{index}.txt")
```

Comment:
super small nitpick - let's do a little left padding on the index so we don't have to worry about the sort order.

Consider using:

`candidate_path = path.with_name(f"{path.stem}_{index:02}.txt")`

Reply:
added